x/reward/keeper/invariants_test.go

```diff
@@ -0,0 +1,62 @@
+package keeper_test
+
+import (
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+
+	testkeeper "github.com/tendermint/spn/testutil/keeper"
+	"github.com/tendermint/spn/testutil/sample"
+	"github.com/tendermint/spn/x/reward/keeper"
+	"github.com/tendermint/spn/x/reward/types"
+)
+
+func TestInsufficientRewardsBalanceInvariant(t *testing.T) {
+	ctx, tk, _ := testkeeper.NewTestSetup(t)
+	t.Run("valid case", func(t *testing.T) {
+		for i := uint64(0); i < uint64(10); i++ {
+			pool := sample.RewardPool(r, i)
+			tk.MintModule(ctx, types.ModuleName, pool.RemainingCoins)
+			tk.RewardKeeper.SetRewardPool(ctx, pool)
+		}
+		msg, broken := keeper.InsufficientRewardsBalanceInvariant(*tk.RewardKeeper)(ctx)
+		require.False(t, broken, msg)
+	})
+
+	t.Run("invalid case 1", func(t *testing.T) {
+		// add some valid pools
+		for i := uint64(0); i < uint64(10); i++ {
+			pool := sample.RewardPool(r, i)
+			tk.MintModule(ctx, types.ModuleName, pool.RemainingCoins)
+			tk.RewardKeeper.SetRewardPool(ctx, pool)
+		}
+		// add some invalid pools - mint a bit less for some coins
+		for i := uint64(10); i < uint64(20); i++ {
+			pool := sample.RewardPool(r, i)
```

Comment:
With `sample.RewardPool` we have random denoms for every reward pools.

To improve the test of the invariant I think we should reuse the same denom between rewards pool so that we check the balance must be greater than sum of coins in reward pools